pkgs/by-name/ut/ut1999/package.nix

```diff
@@ -0,0 +1,122 @@
+{ lib, stdenv, requireFile, autoPatchelfHook, fetchurl,
+libgcc, wxGTK32, innoextract, makeWrapper,
+libGL, SDL2, openal, libmpg123, libxmp }:
+
+let
+  oldUnrealPatch = stdenv.mkDerivation {
+    name = "ut1999-oldunreal-patch";
+    version = "469d";
+    sourceRoot = ".";
+    src = fetchurl {
+      url = "https://github.com/OldUnreal/UnrealTournamentPatches/releases/download/v469d/OldUnreal-UTPatch469d-Linux-amd64.tar.bz2";
+      hash = "sha256-aoGzWuakwN/OL4+xUq8WEpd2c1rrNN/DkffI2vDVGjs=";
+    };
+
+    buildInputs = [
+      libgcc
+      stdenv.cc.cc
+      wxGTK32
+    ];
+
+    nativeBuildInputs = [
+      autoPatchelfHook
+      makeWrapper
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      ls -alh
+      cp -r ./* $out
+
+      runHook postInstall
+    '';
+
+    dontBuild = true;
+  };
+
+  unpackGog = stdenv.mkDerivation {
+    name = "ut1999-gog";
+    version = "2.0.0.5";
+
+    src = requireFile rec {
+      name = "setup_ut_goty_2.0.0.5.exe";  # FIXME: use official name
+      sha256 = "00v8jbqhgb1fry7jvr0i3mb5jscc19niigzjc989qrcp9pamghjc";
+      message = ''
+        Unreal Tournament 1999 requires the official GOG package, version 2.0.0.5.
+
+        Once you download the file, run the following command:
+
+        nix-prefetch-url file://\$PWD/${name}
+      '';
+    };
+
+    buildInputs = [
+      innoextract
+    ];
+
+    unpackPhase = ''
+      innoextract --extract --exclude-temp "$src"
```

Comment:
If you know how this works/can be used, it'd be great to have a generic setup-hook.sh for this package.

Reply:
@Atemu , it's a good point.
Actually I tried to use the `gog-unpack` set up hook from @peterhoeg .

The problem is that the script also does a:

```sh
find . -depth -print -execdir rename -f 'y/A-Z/a-z/' '{}' \;
```

This renames all the case sensitive directories such as `System` ,`Textures` to `system`, `textures`. Theoretically I could account for that when copying the files, but it felt like a hack. Not sure why the gog hook does that.

Reply:
Could be because of assumed case-insensitivity.

If that's the reason, it should be optional.

Reply:
@Atemu , personally I would remove the `find` command from `gog-unpack` and put it into the `dxx-rebirth` (which is the only package that uses this set up hook).

The problem is that (at least for the moment) I cannot test if I break anything since I don't own Descent 1 & 2 on Gog.

Reply:
Yes, that should absolutely be optional. Anyone feels like conjuring up a PR?

Reply:
@Atemu , @peterhoeg , I would give it a try in a separate PR.
@Atemu , do you have an example of a setup hook with optional features?